### ISSUE TITLE
Set custom default state for "collapse repo files" feature

### ISSUE THREAD

**Issue by ZaLiTHkA** [refined-github/refined-github#5979]
### Description

I really like having the file list collapsed while browsing projects on GitHub, but I keep getting caught out with the way it "persists" the last known state across all page (re)loads.

could this be changed so that the file list is collapsed or expanded based on a custom setting, rather than the last known state?

---

regarding the implementation of this change, I believe it would be best if the `toggle-files-button` feature could be "turned on" by a select list, rather than a checkbox, including the following 3 options:

* "Last Known" - which would retain the current beaviour
* "Collapsed" - which would collapse by default on load
* "Expanded" - which would expand by default on load

### Example URLs

https://github.com/refined-github/refined-github

**Comment by fregante** [comment]
Unfortunately that qualifies as an "option", which we don't support

https://github.com/refined-github/refined-github/wiki/%22Can-you-add-this-feature%3F%22#3-it-doesnt-require-options

**Comment by ZaLiTHkA** [comment]
that's unfortunate.. it really is annoying though, especially since the list of files only collapses _after_ the page has finished loading. over slower internet connections (which is pretty much anything affordable here in South Africa), this often results in mis-clicks and/or having to scroll back _up_ after the list is collapsed. and even then, only when _any other_ GitHub page had _its own_ file list collapsed.

tl;dr: collapsing this file list really is helpful. not having it function consistently is almost equally unhelpful.

would it be possible to change the behaviour to "collapsed" by default then, rather than tracking the last known state?

**Comment by fregante** [comment]
> only collapses _after_ the page has finished loading.

Thanks for the report! Fixed in https://github.com/refined-github/refined-github/pull/5980



> would it be possible to change the behaviour to "collapsed" by default then

We had the exact opposite request before 😅

**Comment by ZaLiTHkA** [comment]
> We can't just hide the file list by default, that's mostly what the Home tab is used for.

yup, that does make sense. no worries. (:

> You might like this other extension to make the list shorter: [sindresorhus/hide-files-on-github](https://github.com/sindresorhus/hide-files-on-github)

I'll check that out, thank you.. 👍🏼
